addons/mail/models/models.py

```diff
@@ -46,6 +46,8 @@ def _mail_track(self, tracked_fields, initial):
                     tracking_sequence = 100
                 tracking = self.env['mail.tracking.value'].create_tracking_values(initial_value, new_value, col_name, col_info, tracking_sequence, self._name)
                 if tracking:
+                    if tracking['field_type'] == 'monetary':
+                        tracking['currency_id'] = getattr(self, col_info.get('currency_field', ''), self.company_id.currency_id).id
```

Comment:
We could first check that currency field is tracked, in which case we should have its value available. If not fallback on your computation.

I don't think we should force currency to be tracked but we could benefits from its tracking.

Reply:
The goal of checking if the currency field is tracked is to get the old currency. But this means that we will need another field in the mail tracking value model to keep this currency, no ?

Reply:
Currency change is globally never going to happen as this is an advanced niche case. Adding a second column just to change currency is too much in my opinion. Basically our purpose is to display "300€" and not "300" near your CRM revenues. This is what we want to achieve and a multi currency change on a given record would be displayed like "280 -> 300€" insteasd of "280$ -> 300€"; this is acceptable.

Reply:
Then we don't need to check if the currency field is tracked, because when we check the value of the record, the changes are already saved on the record, so we already know the possible new value of the currency field. The only goal to check if a currency field is tracked is to get the old value not the new.